### PR TITLE
Jobpostings

### PR DESCRIPTION
**Pull request by Marwanyx**
Added job postings controller, model and routes

### CODE REVIEW

File: backend/controllers/jobPostingsController.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    res.status(200).json(jobPosting);
+}
+
+// create a new job posting
+const createJobPosting = async (req, res) => {
+    const { company, jobTitle, location, jobType, details, noDegreeMentioned, benefits } = req.body;
+
+    const newJobPosting = new JobPosting({
```
Comment (stalhaali) [inline backend/controllers/jobPostingsController.js:32]: Should move this after the logic for checking if the same job already exists. Also shouldnt it be possible for two jobs to have the same title?

File: backend/controllers/jobPostingsController.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        res.status(409).json({ message: error.message });
+    }
+}
+
+// delete a job posting by ID
+const deleteJobPosting = async (req, res) => {
+    const { id } = req.params
+
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+        return res.status(404).send('Job posting not found')
```
Comment (stalhaali) [inline backend/controllers/jobPostingsController.js:70]: does this mean the object id does not exist? Or does it mean its invalid? If so then the error response doesnt match what it does. Same for line 75

File: backend/controllers/jobPostingsController.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+// update a job posting by ID
+const updateJobPosting = async (req, res) => {
+    const { id } = req.params
+    const { company, jobTitle, location, jobType, details, noDegreeMentioned, benefits } = req.body;
+
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+        return res.status(404).send('Job posting not found')
+    }
+
```
Comment (stalhaali) [inline backend/controllers/jobPostingsController.js:90]: should also check if job exists just like in deletejobposting function

**Review by tsids — APPROVED** [review]
Fix the merge conflict, and it should be good to go.

**Pull request merged**